modules/effects/schematics-core/utility/visit-utils.ts

```diff
@@ -10,24 +10,35 @@ export function visitTSSourceFiles<Result = void>(
   ) => Result | undefined
 ): Result | undefined {
   let result: Result | undefined = undefined;
+  for (const sourceFile of visit(tree.root)) {
+    result = visitor(sourceFile, tree, result);
+  }
 
-  tree.visit(path => {
-    if (!path.endsWith('.ts')) {
-      return;
-    }
-
-    const sourceFile = ts.createSourceFile(
-      path,
-      tree.read(path)!.toString(),
-      ts.ScriptTarget.Latest
-    );
+  return result;
+}
 
-    if (sourceFile.isDeclarationFile) {
-      return;
+function* visit(directory: DirEntry): IterableIterator<ts.SourceFile> {
```

Comment:
Don't use `tree.visit` for more flexibility (like ignore `node_modules`).
Inspiration from https://github.com/angular/angular-cli/blob/576119f2cab6a53f0bc67026b723fffbe8407258/packages/schematics/angular/migrations/update-8/update-lazy-module-paths.ts. I like this solution the most as we break out the `node_modules` folder as fast as possible.

Another option is to use the `ignore` package, like the nx workspace does - https://github.com/nrwl/nx/blob/3af31f1cd996db2f225773eb1888bd109efb9f76/packages/schematics/migrations/update-8-0-0/update-8-0-0.ts#L176

Or to simple check if `node_modules` is in the path.